core/src/main/scala/io/projectglow/vcf/TabixIndexHelper.scala

```diff
@@ -489,10 +489,10 @@ object TabixIndexHelper extends GlowLogging {
               }
             val blockRangeStart = Math.max(file.start, minOverBlocks >> 16) // Shift 16 bits to get
             // file offset of the bin from bgzipped virtual file offset.
-            val blockRangeEnd = Math.min(file.start + file.length, (maxOverBlocks >> 16) + 0xFFFF)
+            val blockRangeEnd = Math.min(file.start + file.length, maxOverBlocks >> 16)
             // 0xFFFF is the maximum possible length of an uncompressed bin.
             if (blockRangeStart <= blockRangeEnd) {
-              Some((blockRangeStart, blockRangeEnd))
+              Some((blockRangeStart, Math.min(file.start + file.length, blockRangeEnd + 0xFFFF)))
             } else {
               None
             }
```

Comment:
Can you let me know what changes with this fix? It seems to me the final result at the end of line 498 will be the same before and after your changes. I could not immediately see this in your unit tests either.

Reply:
In the unit tests, `r3` is empty because the partitioned file associated with it (`p3`) starts after the beginning of the last indexed block. Previously, `r3` would have been nonempty as the partitioned file starts within the 0xFFFF margin of the last indexed block.